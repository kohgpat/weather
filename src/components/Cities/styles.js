import styled from "styled-components";

export const Cities = styled.div`
  margin-top: calc(50px + var(--topbar-height));
  padding: 20px;

  @supports (display: grid) {
    display: grid;
    grid-template-columns: repeat(1, 1fr);
    grid-gap: 30px;
    grid-auto-rows: minmax(300px, auto);

    @media (min-width: 768px) {
      grid-template-columns: repeat(2, 1fr);
    }

    @media (min-width: 1028px) {
      grid-template-columns: repeat(3, 1fr);
    }
  }

  @supports not (display: grid) {
    display: flex;
    flex-direction: row;
    flex-shrink: 0;
    align-items: center;
    justify-content: center;
    width: 100%;
    flex-wrap: wrap;
  }
`;

export const NoCitiesMessage = styled.div`
  margin-top: calc(50px + var(--topbar-height));
  padding: 20px;
`;
