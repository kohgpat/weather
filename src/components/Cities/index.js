import React from "react";
import City from "../City";
import * as styles from "./styles";

const Cities = ({ cities = [] }) => {
  if (!cities.length) {
    return null;
  }

  return (
    <styles.Cities>
      {cities.map(city => (
        <City key={city.id} city={city} />
      ))}
    </styles.Cities>
  );
};

export default Cities;
