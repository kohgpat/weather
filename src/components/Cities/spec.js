import React from "react";
import Cities from "./index";
import { shallow, mount } from "enzyme";

it("renders", () => {
  const cities = shallow(<Cities />);
  expect(cities).toMatchSnapshot();
});

it("renders - no cities", () => {
  const cities = mount(<Cities />);
  expect(cities.text()).toContain(
    "You have no any cities. Please add one using form below."
  );
});

it("renders - with cities", () => {
  const cities = shallow(
    <Cities cities={[{ id: 1, name: "Moscow" }, { id: 2, name: "Paris" }]} />
  );
  expect(cities.length).toEqual(1);
});
