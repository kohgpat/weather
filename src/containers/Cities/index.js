import React, { Component } from "react";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import * as citiesActions from "../../store/cities/actions";
import * as citiesSelectors from "../../store/cities/selectors";
import Cities from "../../components/Cities";

class CitiesContainer extends Component {
  componentDidMount() {
    this.props.actions.cities.citiesAll();
  }

  handleRemoveCity = (city) => {
    this.props.actions.cities.citiesRemove(city);
  };

  render() {
    const { cities } = this.props;

    return <Cities cities={cities} onRemoveCity={this.handleRemoveCity} />;
  }
}

const mapStateToProps = ({ cities }) => ({
  cities: citiesSelectors.getCities(cities)
});

const mapDispatchToProps = dispatch => ({
  actions: {
    cities: bindActionCreators(citiesActions, dispatch)
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(CitiesContainer);
