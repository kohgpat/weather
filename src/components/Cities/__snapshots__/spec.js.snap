// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<styled.div>
  You have no any cities. Please add one using form below.
</styled.div>
`;
